include/alpaka/warp/Traits.hpp

```diff
@@ -150,5 +155,32 @@ namespace alpaka
             using ImplementationBase = concepts::ImplementationBase<ConceptWarp, TWarp>;
             return traits::Ballot<ImplementationBase>::ballot(warp, predicate);
         }
+
+        //-----------------------------------------------------------------------------
+        //! Broadcasts data from one thread to all members of the warp.
+        //! Similar to MPI_Bcast, but using srcLane instead of root.
+        //!
```

Comment:
IMO we need to add to the documentation that this function `shfl` is collective what means all threads need to call the function and also from the same code branch.
The reason is that for CUDA the implementation is using `activemask` and for HIP all threads in a warp needs to call the function. Using `activemask` means if threads from the if and else branch call the function they will not see each other.

Reply:
I updated these docs to include this warning.

Reply:
I think I forgot to add a similar warning to the previously existing warp collectives. You comment also alllies to those, right @psychocoderHPC ?

Reply:
@sbastrakov Yes this should be added to other warp functions too. Currently, only CUDA allows calling warp functions from different branches. It is fine if all threads of the warp are in the same branch but as soon as the threads diverge the behavior is undefined (for HIP and CUDA devices before sm_70) .